app/bundles/SmsBundle/Entity/SmsRepository.php

```diff
@@ -32,7 +32,12 @@ public function getEntities(array $args = [])
         return parent::getEntities($args);
     }
 
-    public function getPublishedBroadcasts($id = null): \Doctrine\ORM\Internal\Hydration\IterableResult
+    /**
+     * @param numeric|null $id
+     *
+     * @return iterable<Sms>|\Doctrine\ORM\Internal\Hydration\IterableResult<Sms>
+     */
+    public function getPublishedBroadcasts($id = null, bool $oldImplementation = true): \Doctrine\ORM\Internal\Hydration\IterableResult|iterable
```

Comment:
It would be cleaner to create a new method with the new implementation and deprecate the old method.

Reply:
Please check the implementation.